setup.sh

```diff
@@ -1,21 +1,33 @@
 #!/bin/bash
 #
-# Sets up parameters for use with other scripts.  Should be run once.
+# Sets up parameters for use with other scripts.  Removes an instance of param.sh if there exists. 
 # Sample usage: bash setup.sh
-
+# Can be run for any number of times
+rm -r params.sh
 echo "First, choose the resource identifier that specifies your cluster resoure. We
 will set up this name in your ssh configuration, and use it to reference the resource (sherlock)."
 echo
 read -p "Resource identifier (default: sherlock) > "  RESOURCE
 RESOURCE=${RESOURCE:-sherlock}
+if [[ "${RESOURCE}" == "sherlock" ]]
+then
+   SHERLOCK=true
```

Comment:
This is the right idea, but we should have variables that explicitly control different functionality. In the examples you showed above I think we want:

 - USE_LSOF: for the first example (the two commands where one uses lsof)

And then this bit

```
 if $SHERLOCK
    then 
       echo "ssh -L $PORT:localhost:$PORT ${RESOURCE} ssh -L $PORT:localhost:$PORT -N $MACHINE &"
       ssh -L $PORT:localhost:$PORT ${RESOURCE} ssh -L $PORT:localhost:$PORT -N "$MACHINE" &
    else
       echo "ssh rice.stanford.edu -l $FORWARD_USERNAME -K -L  $PORT:$MACHINE:$PORT -N  &"
       ssh rice.stanford.edu -l $FORWARD_USERNAME -K -L  $PORT:$MACHINE:$PORT -N  &
    fi
```

If there are two things here - the first being "do we want to ssh directed to a domain url instead of localhost" (the second in the block) and the second being the URL for that (rice.stanford.edu). I would also do a check in params.sh that the second variable is defined given that the first condition is true.

Reply:
Ah got it - btw, how did you manage to get that double ssh tunnel working for sherlock? The double forwarding does not work for farmshare. 

I think its safer that we ssh to the domain name because I am not sure if all HPCs will have the same idea like sherlock, but definitely will allow ssh to the domain url over localhost, and then we bind via the tunnel. 

I will add a corresponding prompt for the domain name in the setup file, and update the this ssh accordingly as well as the lsof. To be fair - I am not even sure how the end.sh will generalise well to other domains - individuals will have to go in and figure out where the lsof is located - but I can add that to the setup.

Reply:
This was designed years ago, so it would be believable if that technique didn't work anymore. If you'd like test what you are doing for farmshare on sherlock (and it works) I'd be good to update that command to be the same between the two.

I agree that different clusters will have their quirks, and the logic for the quirk-iness is best in that one file to generate params over being scattered all over the code.